lib/recipient.rb

```diff
@@ -0,0 +1,45 @@
+require "pry"
+require "httparty"
+
+class Recipient
+  BASE_URL = "https://slack.com/api/"
+
+  attr_reader :slack_id, :name
+
+  def initialize(slack_id:, name:)
+    @slack_id = slack_id
+    raise ArgumentError if !name.is_a? String
+    @name = name
+  end
+
+  def send_message(params)
+    endpoint = "chat.postMessage"
+    url = BASE_URL + endpoint
+    params[:token] = ENV["SLACK_API_TOKEN"]
+    response = HTTParty.post(url, body: params)
+    unless response.code == 200 && response.parsed_response["ok"]
+      raise SlackApiError, response["error"]
+    end
```

Comment:
Line 21 raises an exception, but not the one you want.

```
select user
You chose to select a user. Please provide a username or Slack ID
dan
Traceback (most recent call last):
	5: from lib/slack.rb:105:in `<main>'
	4: from lib/slack.rb:40:in `main'
	3: from lib/slack.rb:40:in `new'
	2: from /Users/droberts/Ada/c11/projects/slack-cli/lib/workspace.rb:11:in `initialize'
	1: from /Users/droberts/Ada/c11/projects/slack-cli/lib/user.rb:16:in `list'
/Users/droberts/Ada/c11/projects/slack-cli/lib/recipient.rb:33:in `get': uninitialized constant Recipient::SlackApiError (NameError)
```

Ruby doesn't know what a `SlackApiError` is, so you get a `NameError` when you try to raise one. Looks like you're missing `require_relative 'apierror'` at the top of this file.